### ISSUE TITLE
Unable to access bookinfo via gateway

### ISSUE THREAD

**Issue by linsun** [istio/istio#23339]
**Bug description**

With 1.6 beta.0 build, I'm not able to access bookinfo via gateway ip/productpage.  I used default profile with values-istio-multicluster-primary.yaml file and i disabled meshExpansion.enabled to simply config.

Debugged with @howardjohn and he found out that my listener is listening on port 80 while it should be 8080 as what has been configured in istio ingress gw service object.

```
$ istioctl pc listener istio-ingressgateway-64675f94df-bg9ns.istio-system
ADDRESS     PORT      TYPE
0.0.0.0     80        HTTP
0.0.0.0     15090     HTTP
```

k8s ingress gw service:
```
- name: http2
    nodePort: 30237
    port: 80
    protocol: TCP
    targetPort: 8080
```

output localhost:8080/debug/edsz?proxyID=ingress
```
{
  "clusterName": "outbound|80||istio-ingressgateway.istio-system.svc.cluster.local",
  "endpoints": [
    {
      "locality": {
        "region": "us-east",
        "zone": "wdc06"
      },
      "lbEndpoints": [
        {
          "endpoint": {
            "address": {
              "socketAddress": {
                "address": "172.30.95.34",
                "portValue": 8080
              }
            }
          },
          "loadBalancingWeight": 1
        }
      ],
      "loadBalancingWeight": 1
    }
  ]
},
```
Workaround:  edit the ingress gateway service object with kubectl cmd and change target port from 8080 to 80.  This fixed the issue manually.

**Expected behavior**

**Steps to reproduce the bug**

**Version (include the output of `istioctl version --remote` and `kubectl version` and `helm version` if you used Helm)**
1.6 beta0
**How was Istio installed?**
see above

**Environment where bug was observed (cloud vendor, OS, etc)**
IKS 1.16

**Comment by linsun** [comment]
@esnible @sdake @GregHanson @ericvn FYI, curious if you hit the same issue with IKS as well.

**Comment by howardjohn** [comment]
@dgn any ideas here? Looking at the code I don't see how we are sending the wrong listener config for this

**Comment by sdake** [comment]
@linsun I have not run into this issue. I use my own gear.

My tests were very basic:
```
sdake@beast-01:~/try_Istio/istio-1.6.0-alpha.2$ bin/istioctl install --set profile=demo
...
✔ Installation complete
sdake@beast-01:~/try_Istio/istio-1.6.0-alpha.2$ kubectl label namespace default istio-injection=enabled
namespace/default labeled
sdake@beast-01:~/try_Istio/istio-1.6.0-alpha.2$ kubectl apply -f samples/bookinfo/platform/kube/bookinfo.yaml
service/details created
...
sdake@beast-01:~/try_Istio/istio-1.6.0-alpha.2$ kubectl apply -f samples/bookinfo/networking/bookinfo-gateway.yaml
gateway.networking.istio.io/bookinfo-gateway created
virtualservice.networking.istio.io/bookinfo created
sdake@beast-01:~/try_Istio/istio-1.6.0-alpha.2$ kubectl get svc istio-ingressgateway -n istio-system -o jsonpath='{.status.loadBalancer.ingress[0].ip}'
192.168.7.150
```

Opening a web browser to https://192.168.7.150/productpage yields this:
<img width="1098" alt="Screen Shot 2020-04-28 at 11 55 31 PM" src="https://user-images.githubusercontent.com/755849/80568604-37b1c980-89ac-11ea-91e5-3a65e2a3f08c.png">


Cheers
-steve

**Comment by sdake** [comment]
Attempting the workflow you mentioned also works for me if I comment out `values.security`. It appears the `values.security` field is no longer valid.

My IOP is:

```
sdake@beast-01:~/try_Istio/istio-1.6.0-alpha.2$ cat /tmp/blob.yaml
apiVersion: install.istio.io/v1alpha1
kind: IstioOperator
spec:
  # https://istio.io/docs/setup/install/multicluster/shared/#main-cluster
  values:
    # selfSigned is required if Citadel is enabled, that is when values.global.istiod.enabled is false.
#    security:
#      selfSigned: false

    global:
      multiCluster:
        # unique cluster name, must be a DNS label name
        clusterName: main0
      network: network1

      # Mesh network configuration. This is optional and may be omitted if
      # all clusters are on the same network.
      meshNetworks:
        network1:
          endpoints:
          # Always use Kubernetes as the registry name for the main cluster in the mesh network configuration
          - fromRegistry: Kubernetes
          gateways:
          - registry_service_name: istio-ingressgateway.istio-system.svc.cluster.local
            port: 443

        network2:
          endpoints:
          - fromRegistry: remote0
          gateways:
          - registry_service_name: istio-ingressgateway.istio-system.svc.cluster.local
            port: 443

      # Use the existing istio-ingressgateway.
      meshExpansion:
        enabled: false
```

Workflow is:
```
2095  bin/istioctl install --filename /tmp/blob.yaml --set profile=demo
 2096  kubectl get pods -n istio-system
 2097  kubectl get pods
 2098  kubectl label namespace default istio-injection=enabled
 2099  kubectl apply -f samples/bookinfo/platform/kube/bookinfo.yaml
 2100  kubectl apply -f samples/bookinfo/networking/bookinfo-gateway.yaml
 2101  kubectl get svc istio-ingressgateway -n istio-system -o jsonpath='{.status.loadBalancer.ingress[0].ip}'
```

Yields:

<img width="769" alt="Screen Shot 2020-04-29 at 12 13 10 AM" src="https://user-images.githubusercontent.com/755849/80569685-48633f00-89ae-11ea-9a9f-d6247a225788.png">

Cheers
-steve

**Comment by dgn** [comment]
> @dgn any ideas here? Looking at the code I don't see how we are sending the wrong listener config for this

I don't see any obvious connection. I'll reserve an hour this afternoon to take a closer look.

**Comment by linsun** [comment]
@sdake thanks for reporting back!  Yes, I fixed the security blob in the example in a PR (merged now) but unfortunately didn't make to beta.0.  Sorry I forgot that detail earlier.

**Comment by howardjohn** [comment]
I think we were unable to reproduce this, presumably an issue fixed in 1.6.0 or installation issues that occurred
